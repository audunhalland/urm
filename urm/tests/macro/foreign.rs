pub struct Foo;
pub struct Bar;

#[urm::table("foo")]
impl Foo {
    fn id() -> String;
    fn bar_id() -> String;

    #[foreign(Self(id) => Self(id))]
    fn self_to_self() -> Foo;

    #[foreign(Bar(id) => Bar(id))]
    fn bar_to_bar() -> Bar;
}

#[urm::table("bar")]
impl Bar {
    fn id() -> String;
    fn foo_id() -> String;
}

fn main() {}
