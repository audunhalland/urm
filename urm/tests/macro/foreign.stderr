error: Foreign self not supported (yet?)
 --> tests/macro/foreign.rs:9:15
  |
9 |     #[foreign(Self(id) => Self(id))]
  |               ^^^^

error: No Self(..) found
  --> tests/macro/foreign.rs:12:15
   |
12 |     #[foreign(Bar(id) => Bar(id))]
   |               ^^^
